modules/src/food.py

```diff
@@ -0,0 +1,93 @@
+        
+        name = ''
+        type = ''
+
+        if 'restaurant_name' in entities:
+            name = entities['restaurant_name'][0]['value']
+            search_url += '&q={name}'.format(name=name)
+
+        if 'cuisine' in entities:
+            type = entities['cuisine'][0]['value']
+            search_url += '&cuisines={type}'.format(type=type)
+
+        query = requests.get(search_url,headers=header).json()['restaurants']
+
+        if name == '':
+            rand_restaurant = query[random.randrange(0,len(query))]
+
+            template = TextTemplate('Name: ' + rand_restaurant['name'] +
+                                    '\nAddress: ' + rand_restaurant['address'] +
+                                    '\nCuisine: ' + rand_restaurant['cuisines'] +
+                                    '\nAverage Price for Two: ' + rand_restaurant['average_cost_for_two'])
+            text = template.get_text()
+            template = ButtonTemplate(text)
+            template.add_web_url('Restaurant Link', rand_restaurant['url'])
+
+            output['input'] = input
+            output['output'] = template.get_message()
+            output['success'] = True
+        else:
+            restaurant_list = []
+            most_relevant = 0
+            search_success = False
+            for restaurant in query:
+                rest_name = restaurant['name']
+                relevance = difflib.SequenceMatcher(None,name,rest_name).ratio()
+                if relevance >= .75:
+                    search_success = True
+                    if relevance >= most_relevant:
+                        restaurant_list.insert(0,restaurant)
+                        most_relevant = relevance
+            if not search_success:
+                template = TextTemplate('I couldn\'t find any restaurants matching your query. Please try asking me something else.')
+                output['input'] = input
+                output['output'] = template.get_message()
+                output['success'] = False
+            else:
+                out_rest = restaurant_list[0]
+                template = TextTemplate('Name: ' + out_rest['name'] +
+                                        '\nAddress: ' + out_rest['address'] +
+                                        '\nCuisine: ' + out_rest['cuisines'] +
+                                        '\nAverage Price for Two: ' + out_rest['average_cost_for_two'])
+                
+                text = template.get_text()
+                template = ButtonTemplate(text)
+                template.add_web_url('Restaurant Link', out_rest['url'])
+                
+                output['input'] = input
+                output['output'] = template.get_message()
+                output['success'] = True
+
+    except:
```

Comment:
This is unbearable exception, do we know what kind of exception are we catching ?

Reply:
@swapagarwal  I saw other modules code as well and try and except used seems to be very unbounded and is not helpful.And here @finleylau is trying to make a module here with the same implementation.Would you be able to look at it, please?